### ISSUE TITLE
[2022.05] PHP Fatal Errors

### ISSUE THREAD

**Issue by tobiasd** [friendica/friendica#11216]
Please report here any PHP Fatal error message obtained running the 2022.02 stable or 2022.05-dev versions.

Replaces https://github.com/friendica/friendica/issues/10756

**Comment by ghost** [comment]
Hi! 

Lots of errors like those: 
```
2022/02/07 20:08:41 [error] 977015#977015: *9733312 FastCGI sent in stderr: "PHP message: PHP Notice:  Trying to access array offset on value of type bool in /var/www/net/nerdica.net/friendica/src/Module/Photo.php on line 291" while reading response header from upstream, client: 31.17.252.155, server: nerdica.net, request: "GET /photo/contact/48/a85d7459-1861-e3a9-d8ca-99b793773228?ts=1643521273 HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "nerdica.net", referrer: "https://nerdica.net/contact/109593"
2022/02/07 20:08:47 [error] 977015#977015: *9733312 FastCGI sent in stderr: "PHP message: PHP Notice:  Trying to access array offset on value of type bool in /var/www/net/nerdica.net/friendica/src/Module/Photo.php on line 291" while reading response header from upstream, client: 31.17.252.155, server: nerdica.net, request: "GET /photo/contact/48/a85d7459-1861-e3a9-d8ca-99b793773228?ts=1643521273 HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "nerdica.net", referrer: "https://nerdica.net/contact/1400298"
2022/02/07 20:09:02 [error] 977015#977015: *9733312 FastCGI sent in stderr: "PHP message: PHP Notice:  Trying to access array offset on value of type bool in /var/www/net/nerdica.net/friendica/src/Module/Photo.php on line 291" while reading response header from upstream, client: 31.17.252.155, server: nerdica.net, request: "GET /photo/contact/48/a85d7459-1861-e3a9-d8ca-99b793773228?ts=1643521273 HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php7.4-fpm.sock:", host: "nerdica.net", referrer: "https://nerdica.net/contact/109593/posts"
```

**Comment by ghost** [comment]
moved to PHP-notices... anyway, the result is an authentication error...

**Comment by AlfredSK** [comment]
These messages are PHP *notices* - no fatal errors. So, this is the wrong issue.

**Comment by thystra** [comment]
I am migrating my server from one host to another.  I performed update via GIT to latest release and the moved it.  When I attempt to start the server on the new host, I get the following errors.  The static/dbstructure.config.php file is present in the system.  
```
PHP Fatal error:  Uncaught Exception: Missing database structure config file static/dbstructure.config.php in src/Database/DBStructure.php:306
Stack trace:
#0 src/Database/Database.php(87): Friendica\Database\DBStructure::definition()
#1 [internal function]: Friendica\Database\Database->__construct()
#2 vendor/level-2/dice/Dice.php(132): ReflectionMethod->invokeArgs()
#3 vendor/level-2/dice/Dice.php(96): Dice\Dice->Dice\{closure}()
#4 vendor/level-2/dice/Dice.php(247): Dice\Dice->create()
#5 vendor/level-2/dice/Dice.php(149): Dice\Dice->Dice\{closure}()
#6 vendor/level-2/dice/Dice.php(96): Dice\Dice->Dice\{closure}()
#7 bin/console.php(36): Dice\Dice->create()
#8 {main}
  thrown in src/Database/DBStructure.php on line 306
```

**Comment by annando** [comment]
@thystra this message is intentionally thrown when the configuration file is not readable. Please check whether the file `/var/www/friendica.argentwolf.org/html/static/dbstructure.config.php` does exist and that it can be accessed by both the web server process and the command line process.

**Comment by rickgardner077** [comment]
I installed the stable branch on a fresh server via git. The frontend does not load, throws status 500. PHP log reports:

`PHP Fatal error:  Uncaught TypeError: Argument 3 passed to Friendica\Database\Database::__construct() must implement interface PsrExt\Log\LoggerInterface, instance of Friendica\Core\Logger\Factory\Logger given in /var/www/virtual/rick6/friendica/src/Database/Database.php:76"`

**Comment by tobiasd** [comment]
Found this in the `friendica.log` file after a WSOD in the network stream (PHP8.0)

```
index [ERROR]: Uncaught Exception TypeError: "Friendica\Model\Post\Media::splitAttachments(): Argument #4 ($has_media) must be of type bool, null given, called in /src/Model/Item.php on line 2792" at /src/Model/Post/Media.php line 553
{"
exception":"TypeError: Friendica\\Model\\Post\\Media::splitAttachments(): Argument #4 ($has_media) must be of type bool, null given, called in /src/Model/Item.php on line 2792 and defined in /src/ModelPost/Media.php:553
Stack trace:
0 /src/Model/Item.php(2792): Friendica\\Model\\Post\\Media::splitAttachments()
1 /src/Object/Post.php(388): Friendica\\Model\\Item::prepareBody()
2 /src/Object/Thread.php(208): Friendica\\Object\\Post->getTemplateData()
3 /src/Content/Conversation.php(765): Friendica\\Object\\Thread->getTemplateData()
4 /src/Module/Conversation/Network.php(202): Friendica\\Content\\Conversation->create()
5 /src/BaseModule.php(239): Friendica\\Module\\Conversation\\Network->content()
6 /src/App.php(711): Friendica\\BaseModule->run()
7 index.php(44): Friendica\\App->runFrontend()
8 {main}
"} -
 {"file":null,"line":null,"function":null,"uid":"96e03c","process_id":17448}

```

**Comment by annando** [comment]
Should be fixed with one my PRs from this day.

**Comment by tobiasd** [comment]
I've scrolled past the postings of the WSOD occurrence without an error, so I guess your are right :) closing https://github.com/friendica/friendica/issues/11216#issuecomment-1129291999

**Comment by stephanmaus** [comment]
Unfortunately, the Friendica-Guide for uberspace still doesn't work. I've tried to install Friendica on a fresh server and got the error-message "500 Internal Server Error". The frontend still doesn't load. Thanks for all your work!

**Comment by tobiasd** [comment]
Which version of Friendica are you trying to install @stephanmaus if it was the last stable release (2022.03) could you try to install the current release candidate instead (from the 2022.05-rc branch of the git repository)

**Comment by rickgardner077** [comment]
I tried installing the 2022.05-rc. Unfortunately no difference from https://github.com/friendica/friendica/issues/11216#issuecomment-1114967771

**Comment by tobiasd** [comment]
What version of PHP do you have on the server @rickgardner077 ?

I just did a successful installation of Friendica 2022.05-rc on a Debian/Bullseye server with PHP 7.4.

**Comment by stephanmaus** [comment]
> Which version of Friendica are you trying to install @stephanmaus

Yes, it was the last stable release (2022.03). I finally suceeded to install it on my other provider "all-inkl" (Friendica 'Siberian Iris' 2022.03 - 1452) where it runs very well. I don't know why it didn't work on uberspace. 

Thanks a lot for your work for Friendica!

**Comment by rickgardner077** [comment]
> What version of PHP do you have on the server @rickgardner077 ?

7.4

**Comment by rickgardner077** [comment]
I just gave it another try on [Uberspace](https://uberspace.de). Sorry that I can't be of more help, but in this case finding the error is above my knowledge. I cloned 2022.05-rc + addon (rc as well) on an empty Uberspace with PHP 7.4 but run into the same error (Internal Server Error 500).

I again followed[ this guide](https://lab.uberspace.de/guide_friendica/), which is not more than cloning the repo and installing it with composer included in bin/composer.phar. 

Friendica should be at [https://rick6.uber.space](https://rick6.uber.space). I'll leave it as is if you want to look at it.
